container-search/src/main/java/com/yahoo/search/grouping/vespa/ResultBuilder.java

```diff
@@ -173,18 +173,23 @@ Group fill(Group group) {
                     }
                 }
             }
-            for (GroupListBuilder child : childLists) {
-                if (child != null) {
-                    group.add(child.build());
+            if (childLists != null) {
+                for (GroupListBuilder child : childLists) {
+                    if (child != null) {
+                        group.add(child.build());
+                    }
                 }
             }
             return group;
         }
 
         GroupListBuilder getOrCreateChildList(int tag, boolean ranked) {
             int index = tag + 1; // Add 1 to avoid the dreaded -1 default value.
-            if (index >= childLists.length) {
-                childLists = Arrays.copyOf(childLists, tag + 8);
+            if (childLists == null || index >= childLists.length) {
+                int reservedSize = (((index + 1) + (CHILDLIST_SIZE_INCREMENTS -1))/CHILDLIST_SIZE_INCREMENTS) * CHILDLIST_SIZE_INCREMENTS;
```

Comment:
```suggestion
                int reservedSize = ((index + CHILDLIST_SIZE_INCREMENTS) / CHILDLIST_SIZE_INCREMENTS) * CHILDLIST_SIZE_INCREMENTS;
```

Reply:
I added minSize = index + 1 to make it clearer. I wanted to keep the (CHILDLIST_SIZE_INCREMENTS - 1) as together with the division and the multiplication is an often used pattern for rounding of integers.